src/common/util/game.ts

```diff
@@ -652,16 +652,37 @@ export function executeCmd(
 
   state.callbackAfterExecution = undefined;
 
+  state.executionStackDepth += 1;
+  if (state.executionStackDepth >= MAX_EXECUTION_STACK_SIZE) {
+    throw new Error('EXECUTION_STACK_DEPTH_EXCEEDED');
+  }
+
   const vocabulary = (buildVocabulary as BuildVocabulary)(state, currentObject, source);
   const [terms, definitions] = [Object.keys(vocabulary), Object.values(vocabulary)];
   const wrappedCmd = `(function (${terms.join(',')}) { return (${cmd})(); })`;
 
   try {
-    return eval(wrappedCmd)(...definitions);  // eslint-disable-line no-eval
+    const result = eval(wrappedCmd)(...definitions);  // eslint-disable-line no-eval
```

Comment:
Heh I know this is vital to Wordbots but this is such a huge XSS vulnerability

Reply:
shhh